proposals/001-controlled-rollout.md

```diff
@@ -0,0 +1,115 @@
+# Controlled Rollout proposal
+
+## Why
+
+New features are great, everybody loves them. However, new features come by the hand of new code. New code can have bugs 
+or it can have performance regressions.
+
+Features aren't perfect for all users. Sometimes they are perfect for some users and not others. Sometimes they are 
+perfect for some hosts and not others.
+
+Rolling out a feature shouldn't always be a binary choice: enabled/disabled. In an ideal world, all features would be 
+enabled by default, everybody would love them all, and they would work flawlessly for all possible use cases.
+
+We are in the real world, though, which is not ideal. So we should give people running Fleet tools to rollout features 
+slowly, so that they can update infrastructure if needed, or only use a feature within the scope that is useful for 
+them.
+
+This is a proposal on how this tool could look like and work.
+
+## How
+
+We would create a new type of boolean value in our `AppConfig` called `RolloutBoolean`.
+
+`RolloutBoolean` will have a function `Enabled(h *fleet.Host) bool`. So instead of doing this:
+
+```go 
+if ac.HostSettings.EnableHostUsers {
+    ...
+}
+```
+
+We would do:
+
+```go 
+if ac.HostSettings.EnableHostUsers.Enabled(host) {
```

Comment:
very much a nit, but it would be nice to think about a naming convention for these booleans that reads better than `EnableHostUsers.Enabled`

Reply:
+1 for changing the name of this method, `Enabled` is a little too specific as well. I would propose something generic like `Value` or `Get` to retrieve the actual value of the boolean based on the host.

Reply:
I like `Get`!